Sources/Features/Metrics/Model/BacktraceMetricsSender.swift

```diff
@@ -0,0 +1,91 @@
+import Foundation
+
+final class BacktraceMetricsSender {
+
+    private let api: BacktraceApi
+    private let metricsContainer: BacktraceMetricsContainer
+    private let settings: BacktraceMetricsSettings
+
+    static let defaultBaseUrl = "https://events.backtrace.io/api"
+    private let baseUrl: String
+
+    enum MetricsUrlPrefix: CustomStringConvertible {
+      case summed
+      case unique
+
+      var description: String {
+        switch self {
+        case .summed: return "summed-events"
+        case .unique: return "unique-events"
+        }
+      }
+    }
+
+    init(api: BacktraceApi, metricsContainer: BacktraceMetricsContainer, settings: BacktraceMetricsSettings) {
+        self.api = api
+        self.metricsContainer = metricsContainer
+        self.settings = settings
+        self.baseUrl = BacktraceMetricsSender.defaultBaseUrl
+    }
+
+    func enable() {
+        sendStartupEvents()
+    }
+
+    private func sendStartupEvents() {
+        sendStartupSummedEvent()
+        sendStartupUniqueEvent()
+    }
+
+    private func sendStartupUniqueEvent() {
+        sendUniqueEvent()
+    }
+
+    private func sendStartupSummedEvent() {
+        sendSummedEvent()
+    }
+
+    private func sendUniqueEvent() {
+        let payload = metricsContainer.getUniqueEventsPayload()
+
+        do {
+            let url = try getSubmissionUrl(urlPrefix: MetricsUrlPrefix.unique)
+            let result = try api.sendMetrics(payload, url: url)
+            handleUniqueEventsResult(result: result)
+        } catch {
+            BacktraceLogger.error(error)
+        }
+    }
+
+    private func sendSummedEvent() {
+        let payload = metricsContainer.getSummedEventsPayload()
+
+        do {
+            let url = try getSubmissionUrl(urlPrefix: MetricsUrlPrefix.summed)
+            let result = try api.sendMetrics(payload, url: url)
+            handleSummedEventsResult(result: result)
+        } catch {
+            BacktraceLogger.error(error)
+        }
+    }
+
+    func getSubmissionUrl(urlPrefix: MetricsUrlPrefix) throws -> URL {
+        let token = try api.credentials.getSubmissionToken()
+        let universe = try api.credentials.getUniverseName()
+
+        let urlString = self.baseUrl + "/" + urlPrefix.description + "/submit?token=" + token + "&universe=" + universe
```

Comment:
It's fine but I'd suggest using `URLComponents` to create complex URLs. It'll make it easier to understand.

Reply:
I understand and agree in general but I think in this case it becomes more difficult to understand because of all the error handling:

```swift
        let token = try api.credentials.getSubmissionToken()
        let universe = try api.credentials.getUniverseName()

        guard let baseUrl = URL(string: baseUrlString) else {
            throw BacktraceUrlParsingError.invalidInput(baseUrlString)
        }

        guard var components = URLComponents(url: baseUrl, resolvingAgainstBaseURL: true) else {
            throw BacktraceUrlParsingError.invalidInput(baseUrl.debugDescription)
        }

        components.path = urlPrefix.description + "/submit"
        components.queryItems = [
            URLQueryItem(name: "token", value: token),
            URLQueryItem(name: "universe", value: universe)
        ]

        guard let url = components.url else {
            throw BacktraceUrlParsingError.invalidInput(baseUrl.debugDescription)
        }

        return url
```

Reply:
Personally, I am ok with those changes. You're right that there is a bit more code for error handling but it gives a nice overview of what's happening. It's up to you to decide and I'm not going to block PR for this reason 😉

Reply:
I will use the URLComponents method, thank you!